components/skus/renderer/skus_js_handler.cc

```diff
@@ -25,6 +27,32 @@
 
 namespace skus {
 
+namespace {
+void ExecuteScript(blink::WebLocalFrame* web_frame, const std::string script) {
+  if (web_frame->IsProvisional())
+    return;
+
+  web_frame->ExecuteScript(
+      blink::WebScriptSource(blink::WebString::FromUTF8(script)));
+}
+void SetProviderNonWritable(blink::WebLocalFrame* web_frame,
+                            const std::string& provider) {
+  const char* provider_str = provider.c_str();
+  const std::string script = base::StringPrintf(
```

Comment:
What are you trying to do here? You should not need to execute JS like this

Reply:
done, added native implementation